esmvaltool/cmorizers/data/formatters/datasets/esacci_soilmoisture.py

```diff
@@ -0,0 +1,191 @@
+    arguments; if bounds are None they will be fixed regardless.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        data cube with coordinates to be fixed.
+
+    overwrite_time_bounds: bool (optional)
+        set to False not to overwrite time bounds.
+
+    overwrite_lon_bounds: bool (optional)
+        set to False not to overwrite longitude bounds.
+
+    overwrite_lat_bounds: bool (optional)
+        set to False not to overwrite latitude bounds.
+
+    Returns
+    -------
+    cube: iris.cube.Cube
+        data cube with fixed coordinates.
+    """
+    # first fix any completely missing coord var names
+    fix_dim_coordnames(cube)
+    # fix individual coords
+    for cube_coord in cube.coords():
+        # fix time
+        if cube_coord.var_name == 'time':
+            logger.info("Fixing time...")
+            cube.coord('time').convert_units(
+                Unit('days since 1970-01-01T00:00:00+00:00',
+                     calendar='proleptic_gregorian'))
+            if overwrite_time_bounds or not cube.coord('time').has_bounds():
+                fix_bounds(cube, cube.coord('time'))
+
+        # fix longitude
+        if cube_coord.var_name == 'lon':
+            logger.info("Fixing longitude...")
+            if cube_coord.ndim == 1:
+                if cube_coord.points[0] < 0. and \
+                        cube_coord.points[-1] < 181.:
+                    cube_coord.points = \
+                        cube_coord.points + 180.
+                    cube.attributes['geospatial_lon_min'] = 0.
+                    cube.attributes['geospatial_lon_max'] = 360.
+                    nlon = len(cube_coord.points)
+                    roll_cube_data(cube, nlon // 2, -1)
+            if overwrite_lon_bounds or not cube_coord.has_bounds():
+                fix_bounds(cube, cube_coord)
+
+        # fix latitude
+        if cube_coord.var_name == 'lat':
+            logger.info("Fixing latitude...")
+            if overwrite_lat_bounds or not cube.coord('latitude').has_bounds():
+                fix_bounds(cube, cube.coord('latitude'))
+
+    return cube
+
+
+def extract_variable(var_info, raw_info, attrs, year):
+    """Extract variables."""
+    rawvar = raw_info['name']
+    constraint = iris.Constraint(name=rawvar)
+    if rawvar == 'sm_uncertainty':
+        sm_cube = iris.load_cube(raw_info['file'],
+                                 iris.NameConstraint(var_name='sm'))
+        ancillary_var = sm_cube.ancillary_variable(
+            'Volumetric Soil Moisture Uncertainty'
+        )
+        cube = sm_cube.copy(ancillary_var.core_data())
+    else:
+        cube = iris.load_cube(raw_info['file'], constraint)
+
+    # Continue with processing the cube if it was successfully loaded
+    fix_var_metadata(cube, var_info)
+    convert_timeunits(cube, year)
+    fix_coords_esacci_soilmoisture(cube, overwrite_time_bounds=False)
+    set_global_atts(cube, attrs)
```

Comment:
```suggestion
```

You can move this part right before the final cube is saved. Then you also don't need to pass `attrs` to this function

Reply:
I moved the function as suggested in the cmorization function but I had to keep the attrs parameter as it is defined in the utilities:

`        if var_name == 'sm':
            final_cube.attributes["history"] = f"Created on {datetime.now()}"
            set_global_atts(cube, glob_attrs)
            save_variable(final_cube, var_name, out_dir, glob_attrs,
                          unlimited_dimensions=['time'])

            # Calculate and save monthly means with Lmon mip
            monthly_mean_cube = monthly_statistics(final_cube, 'mean')
            monthly_mean_cube.var_name = var_name
            monthly_mean_cube.attributes["history"] = (
                f"Created on {datetime.now()}")
            glob_attrs['mip'] = 'Lmon'
            monthly_mean_cube.attributes.update(glob_attrs)
            set_global_atts(cube, glob_attrs)
            save_variable(monthly_mean_cube, var_name, out_dir, glob_attrs,
                          unlimited_dimensions=['time'])
        else:
            # Save the smStderr data
            final_cube.attributes["history"] = f"Created on {datetime.now()}"
            set_global_atts(cube, glob_attrs)
            save_variable(final_cube, var_name, out_dir, glob_attrs,
                          unlimited_dimensions=['time'])`